### PR TITLE
Bring back `component: velero` label

### PR DESCRIPTION
**Pull request by kaovilai**
fix [OADP-458](https://issues.redhat.com//browse/OADP-458)

### CODE REVIEW

**Comment by JaydipGabani** [comment]
@kaovilai I am a little confused so bear with me, the label we are looking for is `component: velero` and which I don't see in this changes, unless that is somehow a default label that now is going to be present. Would you be able to post velero pod yaml in your PR here to display that the desired label is now present after your changes?

File: controllers/velero.go [hunk 1]
```diff
@@ -392,6 +392,7 @@ func (r *DPAReconciler) buildVeleroDeployment(veleroDeployment *appsv1.Deploymen
 	)
 	veleroDeployment.TypeMeta = installDeployment.TypeMeta
 	veleroDeployment.Spec = installDeployment.Spec
+	veleroDeployment.Labels = installDeployment.Labels
```
Comment (kaovilai) [inline controllers/velero.go:395]: @JaydipGabani This adds component: velero to veleroDeployment

File: controllers/velero.go [hunk 2]
```diff
@@ -427,23 +428,16 @@ func removeDuplicateValues(slice []string) []string {
 }
 
 func (r *DPAReconciler) customizeVeleroDeployment(dpa *oadpv1alpha1.DataProtectionApplication, veleroDeployment *appsv1.Deployment) error {
-	veleroDeployment.Labels = r.getAppLabels(dpa)
```
Comment (kaovilai) [inline controllers/velero.go:430]: We don't override labels

File: controllers/velero.go [hunk 3]
```diff
@@ -427,23 +428,16 @@ func removeDuplicateValues(slice []string) []string {
...
-		MatchLabels: map[string]string{
-			"app.kubernetes.io/name":       common.Velero,
-			"app.kubernetes.io/instance":   dpa.Name,
-			"app.kubernetes.io/managed-by": common.OADPOperator,
-			"app.kubernetes.io/component":  Server,
-			oadpv1alpha1.OadpOperatorLabel: "True",
-		},
+	//append dpa labels
+	for k, v := range r.getDpaAppLabels(dpa) {
+		if veleroDeployment.Labels[k] == "" { //saves component: velero labels
```
Comment (kaovilai) [inline controllers/velero.go:433]: We set if it's not set already 
preserves label from https://github.com/openshift/oadp-operator/pull/638/files#r849903351

File: controllers/restic.go [hunk 4]
```diff
@@ -152,6 +152,7 @@ func (r *DPAReconciler) buildResticDaemonset(dpa *oadpv1alpha1.DataProtectionApp
 	ds.TypeMeta = installDs.TypeMeta
 	// Update Spec
 	ds.Spec = installDs.Spec
+	ds.Labels = installDs.Labels
```
Comment (kaovilai) [inline controllers/restic.go:155]: @JaydipGabani  this solves sets label for restic.

**Review by sseago — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**